internal/service/storagegateway/gateway.go

```diff
@@ -495,6 +496,21 @@ func resourceGatewayRead(ctx context.Context, d *schema.ResourceData, meta any)
 		return diags
 	}
 
+	if isGatewayNotConnectedErr(err) {
+		var gatewayInfo *awstypes.GatewayInfo
+		gatewayInfo, err = findGatewayInfoByARN(ctx, conn, d.Id())
+
+		if err == nil {
+			d.Set(names.AttrARN, gatewayInfo.GatewayARN)
+			d.Set("ec2_instance_id", gatewayInfo.Ec2InstanceId)
+			d.Set("gateway_name", gatewayInfo.GatewayName)
+			d.Set("gateway_type", gatewayInfo.GatewayType)
+			d.Set("host_environment", gatewayInfo.HostEnvironment)
+
+			return diags
+		}
```

Comment:
nit to take or leave - if the original finder `err` is not shadowed, its message can be returned if the fallback fails. I'm not sure what is more helpful for users to see in this case.

```suggestion
		if gatewayInfo, err := findGatewayInfoByARN(ctx, conn, d.Id()); err == nil {
			d.Set(names.AttrARN, gatewayInfo.GatewayARN)
			d.Set("ec2_instance_id", gatewayInfo.Ec2InstanceId)
			d.Set("gateway_name", gatewayInfo.GatewayName)
			d.Set("gateway_type", gatewayInfo.GatewayType)
			d.Set("host_environment", gatewayInfo.HostEnvironment)

			return diags
		}
```